### PR TITLE
CXXCBC-327: bundle Mozilla certificates with the library

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

**Comment by avsej** [comment]
Sample build info output:

```json
{
  "_MSC_VER": "1934",
  "__cplusplus": "199711",
  "asio": "1.24.0",
  "build_timestamp": "2023-05-09 19:05:19",
  "cc": "MSVC 19.34.31944.0",
  "cmake_build_type": "Release",
  "cmake_version": "3.26.3",
  "compile_definitions": "_WIN32_WINNT=0x0601;FMT_LOCALE;SPDLOG_COMPILED_LIB;SPDLOG_FMT_EXTERNAL",
  "compile_features": "cxx_std_17;cxx_variadic_templates",
  "compile_flags": "",
  "compile_options": "/W4;/wd4189;/wd4242;/wd4244;/w14254;/w14263;/w14265;/w14287;/we4289;/w14296;/w14311;/w14545;/w14546;/w14547;/w14549;/w14555;/wd4619;/w14640;/w14826;/w14905;/w14906;/w14928;/permissive-;/WX",
  "cpu": "AMD64",
  "cxx": "MSVC 19.34.31944.0",
  "fmt": "8.800.1",
  "http_parser": "2.9.4",
  "link_depends": "",
  "link_flags": "",
  "link_libraries": "project_options;project_warnings;fmt::fmt;spdlog::spdlog;snappy;jsonsl;hdr_histogram_static;couchbase_backtrace;couchbase_logger;couchbase_platform;couchbase_meta;couchbase_crypto;couchbase_sasl;couchbase_tracing;couchbase_metrics;http_parser;iphlpapi;OpenSSL::SSL;OpenSSL::Crypto",
  "link_options": "",
  "mozilla_ca_bundle_date": "fb1ecd641d0a02c01bc9036d513cb658bbda62a75e246bedbc01764560a639f0",
  "mozilla_ca_bundle_embedded": true,
  "mozilla_ca_bundle_sha256": "Tue Jan 10 04:12:06 2023 GMT",
  "mozilla_ca_bundle_size": 137,
  "openssl_config_dir": "C:\\Program Files\\Common Files\\SSL",
  "openssl_crypto_interface_imported_location": "",
  "openssl_crypto_interface_include_directories": "C:/Program Files/OpenSSL/include",
  "openssl_crypto_interface_link_libraries": "",
  "openssl_default_cert_dir": "C:\\Program Files\\Common Files\\SSL/certs",
  "openssl_default_cert_dir_env": "SSL_CERT_DIR",
  "openssl_default_cert_file": "C:\\Program Files\\Common Files\\SSL/cert.pem",
  "openssl_default_cert_file_env": "SSL_CERT_FILE",
  "openssl_headers": "OpenSSL 1.1.1t  7 Feb 2023",
  "openssl_pkg_config_interface_include_directories": "",
  "openssl_pkg_config_interface_link_libraries": "",
  "openssl_runtime": "OpenSSL 1.1.1t  7 Feb 2023",
  "openssl_ssl_imported_location": "",
  "openssl_ssl_interface_include_directories": "C:/Program Files/OpenSSL/include;C:/Program Files/OpenSSL/include",
  "openssl_ssl_interface_link_libraries": "OpenSSL::Crypto",
  "platform": "Windows-10.0.20348",
  "platform_name": "Windows",
  "platform_version": "10.0.20348",
  "post_linked_openssl": "OFF",
  "revision": "d49907b4b294ef36e3724bf76594f4554092a1f7",
  "semver": "1.0.0+d49907b",
  "snappy": "1.1.10",
  "snapshot": false,
  "spdlog": "1.9.2",
  "static_openssl": false,
  "static_stdlib": false,
  "txns_forward_compat_extensions": "TI,MO,BM,QU,SD,BF3787,BF3705,BF3838,RC,UA,CO,BF3791,CM,SI,QC,IX,TS",
  "txns_forward_compat_protocol_version": "2.0",
  "version": "1.0.0",
  "version_build": 0,
  "version_major": 1,
  "version_minor": 0,
  "version_patch": 0
}
```

**Comment by avsej** [comment]
jenkins, test this please

File: core/meta/version.cxx [hunk 1]
```diff
@@ -112,8 +114,19 @@ sdk_build_info()
         }
     }
 #endif
+
+#if defined(COUCHBASE_CXX_CLIENT_EMBED_MOZILLA_CA_BUNDLE)
+    info["mozilla_ca_bundle_embedded"] = "true";
+    info["mozilla_ca_bundle_sha256"] = COUCHBASE_CXX_CLIENT_MOZILLA_CA_BUNDLE_DATE;
```
Comment (thejcfactor) [inline core/meta/version.cxx:120]: The SHA and BUNDLE_DATE are swapped 🙂

Comment (avsej) [reply core/meta/version.cxx:120]: Thank you. I will fix it

**Review by thejcfactor — APPROVED** [review]
LGTM 👍 

I like the CMake incantation to build cert bundle 🙂

**Pull request merged**